### PR TITLE
Task/2413614 chromatic setup GitHub actions

### PR DESCRIPTION
**Pull request by anjulgarg**
# What
Adds a chromatic.yml github action for publishing to Chromatic on each PR

# Why
For viewing per branch Storybook 
For Visual comparison of component changes

### CODE REVIEW

File: .github/workflows/chromatic.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Install Chromatic
+        run: npm i -g chromatic
+
+      # - name: Publish to Chromatic
+      # working-directory: ./packages/storybook
+      # run: chromatic --project-token=${{ secrets.CHROMATIC_PROJECT_TOKEN }} -b build
+      # 👇 Adds Chromatic as a step in the workflow
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
+        # Chromatic GitHub Action options
```
Comment (prprabhu-ms) [inline .github/workflows/chromatic.yml:35]: Nit: I'd drop this comment.

File: .github/workflows/chromatic.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: 'Chromatic'
```
Comment (prprabhu-ms) [inline .github/workflows/chromatic.yml:1]: nit: I'd name the file and the action 'publish-to-chromatic'

**Review by prprabhu-ms — APPROVED** [review]

File: .github/workflows/chromatic.yml (hunk 1 above)
Comment (anjulgarg) [reply .github/workflows/chromatic.yml:38]: Was still debugging. Gonna clean up the files now ;)

File: .github/workflows/chromatic.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Install Dependencies
+        run: rush install
+
+      - name: Rush Build Storybook
+        run: rush build -T storybook
+
+      - name: Install Chromatic
+        run: npm i -g chromatic
+
+      # - name: Publish to Chromatic
```
Comment (JamesBurnside) [inline .github/workflows/chromatic.yml:32]: nit: drop this also

File: .github/workflows/chromatic.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          node-version: 14.x
+
+      - name: Install Rush
+        run: npm install -g @microsoft/rush@5.47.0
+
+      - name: Install Dependencies
+        run: rush install
+
+      - name: Rush Build Storybook
```
Comment (JamesBurnside) [inline .github/workflows/chromatic.yml:26]: nit: `Build Storybook dependencies`

**Review by JamesBurnside — APPROVED** [review]
This is awesome! ♥

File: .github/workflows/publish-chromatic.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  chromatic_deployment:
+    name: Publish Chromatic
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```
Comment (prprabhu-ms) [inline .github/workflows/publish-chromatic.yml:13]: This could use

    with:
        fetch-depth: 0

to use a shallow checkout.

**Comment by JamesBurnside** [comment]
Do we want to consider not having it on every push of every PR? Just thinking for stateful-only changes, or infra PRs its not necessary.
Some possible options would be:
* Only run if there are changes to [react-composite, react-components, storybook] files
* Only run if there is a label on the PR

(Note: not a change for this PR, just thinking if we will queue a lot of unnecessary chromatic deployments!)

**Comment by JamesBurnside** [comment]
> Storybook URL [60c7ae6891f0e90039d7cd54-acrrfmicgg.chromatic.com/iframe.html](https://60c7ae6891f0e90039d7cd54-acrrfmicgg.chromatic.com/iframe.html)

This is perfect! Should/can we set the storybook deployments to be public by default?

**Review by alcail — APPROVED** [review]

**Comment by anjulgarg** [comment]
> Do we want to consider not having it on every push of every PR? Just thinking for stateful-only changes, or infra PRs its not necessary.
> Some possible options would be:
> 
> * Only run if there are changes to [react-composite, react-components, storybook] files
> * Only run if there is a label on the PR
> 
> (Note: not a change for this PR, just thinking if we will queue a lot of unnecessary chromatic deployments!)

Are you a mind reader?
That is precisely what I am working on next :D

**Comment by anjulgarg** [comment]
> > Storybook URL [60c7ae6891f0e90039d7cd54-acrrfmicgg.chromatic.com/iframe.html](https://60c7ae6891f0e90039d7cd54-acrrfmicgg.chromatic.com/iframe.html)
> 
> This is perfect! Good bot 😄 Should/can we set the storybook deployments to be public by default?

Yes. Made the storybook Public by default. Anyone with the link can view it :)

File: .github/workflows/publish-chromatic.yml (hunk 5 above)
Comment (JamesBurnside) [reply .github/workflows/publish-chromatic.yml:17]: Looks like the chromatic github action asks for this too:
image
https://github.com/chromaui/action

**Comment by anjulgarg** [comment]
> Do we want to consider not having it on every push of every PR? Just thinking for stateful-only changes, or infra PRs its not necessary.
> Some possible options would be:
> 
> * Only run if there are changes to [react-composite, react-components, storybook] files
> * Only run if there is a label on the PR
> 
> (Note: not a change for this PR, just thinking if we will queue a lot of unnecessary chromatic deployments!)

This is done as well.
Publish to Chromatic will only run when there is a code change in said directories.

**Pull request merged**